litestar/app.py

```diff
@@ -450,6 +451,33 @@ def __init__(
             config.stores if isinstance(config.stores, StoreRegistry) else StoreRegistry(config.stores)
         )
 
+    @property
+    def cli_plugins(self) -> list[CLIPluginProtocol]:
+        return self.get_plugins_of_type(CLIPluginProtocol)
+
+    @property
+    def openapi_schema_plugins(self) -> list[OpenAPISchemaPluginProtocol]:
+        return self.get_plugins_of_type(OpenAPISchemaPluginProtocol)
+
+    @property
+    def serialization_plugins(self) -> list[SerializationPluginProtocol]:
+        return self.get_plugins_of_type(SerializationPluginProtocol)
+
+    def get_plugin(self, type_: type[PluginT]) -> PluginT:
+        """Return the registered plugin of ``type_``.
+
+        This should be used with subclasses of the plugin protocols.
+        """
+        try:
+            return cast(PluginT, self._plugins[type_])
+        except KeyError as e:
+            raise KeyError(f"No plugin of type {type_!r} registered") from e
+
+    @lru_cache  # noqa: B019
+    def get_plugins_of_type(self, type_: type[PluginT]) -> list[PluginT]:
+        """Return all registered plugins that are subclasses of ``type_``."""
+        return [p for p in self._plugins.values() if isinstance(p, type_)]
```

Comment:
Ideally this should be an immutable type like a frozenset, but it's not for reasons of backwards compatibility.